test/e2e/pkg/grammar/checker_test.go

```diff
@@ -0,0 +1,103 @@
+package grammar
+
+import (
+	"fmt"
+	"testing"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+)
+
+type Test struct {
+	name         string
+	abciCalls    []*abci.Request
+	isCleanStart bool
+	result       bool
+}
+
+var (
+	initChain       = &abci.Request{Value: &abci.Request_InitChain{InitChain: &abci.RequestInitChain{}}}
+	finalizeBlock   = &abci.Request{Value: &abci.Request_FinalizeBlock{FinalizeBlock: &abci.RequestFinalizeBlock{}}}
+	commit          = &abci.Request{Value: &abci.Request_Commit{Commit: &abci.RequestCommit{}}}
+	offerSnapshot   = &abci.Request{Value: &abci.Request_OfferSnapshot{OfferSnapshot: &abci.RequestOfferSnapshot{}}}
+	applyChunk      = &abci.Request{Value: &abci.Request_ApplySnapshotChunk{ApplySnapshotChunk: &abci.RequestApplySnapshotChunk{}}}
+	prepareProposal = &abci.Request{Value: &abci.Request_PrepareProposal{PrepareProposal: &abci.RequestPrepareProposal{}}}
+	processProposal = &abci.Request{Value: &abci.Request_ProcessProposal{ProcessProposal: &abci.RequestProcessProposal{}}}
+)
+
+const (
+	CLEAN_START = true
+	PASS        = true
+	FAIL        = false
+)
+
+var tests = []Test{
```

Comment:
This is OK, although it tends to be defined inside the test case that uses them, for readability